routes/images.js

```diff
@@ -0,0 +1,35 @@
+const express = require('express')
+const router = express.Router()
+const debug = require('debug')('k8s-autodeploy:services')
+const {executeCommand, deployment} = require('../util/commands')
+const {commonErrorHandler} = require('../util/errors');
+
+// Restart all deployments that run a specific image
+router.post('/:images/rollout', async function (request, response, next) {
+  const dockerTag = request.body.push_data.tag
+  const images = request.params.images.split(',')
+
+  debug('NEW rollout request. Images => %s, DockerTag => %s', request.params.images, dockerTag)
+  try {
+    await Promise.all(
+      images.map(async image => {
+        // Gets all deployments running the given image separated by space
+        const command = `kubectl get deployments -o jsonpath="{range .items[*]}{.metadata.name}{' '}{.spec.template.spec.containers[*].image}{'\\n'}{end}" | grep ${image} | awk '{print $1}' | xargs`
+        const deployments = (await executeCommand('sh', ['-c', command])).trim();
+        debug('Found deployments => %s', deployments)
+        await Promise.all(
+          deployments.split(' ').map(async deploymentName => {
+            await deployment.rollout(deploymentName);
+            debug('Deployment rolled out => %s', deploymentName)
+          })
+        );
+      })
+      );
+      response.status(200).json({})
+    } catch (error) {
```

Comment:
Instead of having this `try/catch` you should be able to call `.catch()` on the promise itself to reduce 1 level of indentation.

Reply:
Personally I prefer

```js
try {
    await promise;
} catch (e) {
    handleError(e);
}
```

Over

```js
await promise.catch(handleError);
```

While the latter is more concise, it does not make it obvious after the `await` which "control flow branch" you took (i.e. was it an error, was it a success) and makes it harder to do stuff in a debugger.

That being said, the latter is more concise and nothing wrong with it either.

I have no strong feeling either way.